snippets/ref-s2i-task.adoc

```diff
@@ -0,0 +1,70 @@
+.Example usage of the `s2i-{lang}` task
+[source,yaml,subs="attributes+"]
+----
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: build-and-deploy
+spec:
+# ...
+  tasks:
+# ...
+  - name: build-s2i
+    taskRef:
+      resolver: cluster
+      params:
+      - name: kind
+        value: task
+      - name: name
+        value: s2i-{lang}
+      - name: namespace
+        value: openshift-pipelines
+    workspaces:
+    - name: source
+      workspace: shared-workspace
+# ...
+----
+
+.Supported parameters for the `s2i-{lang}` task
+[cols="1,2,1,1",options="header"]
+|===
+| Parameter | Description | Type | Default value
+|`IMAGE` |The fully qualified name for the container image that the S2I process builds. |`string` |
+|`IMAGE_SCRIPTS_URL` |The URL containing the default assemble and run scripts for the builder image. |`string` |`image:///usr/libexec/s2i`
+|`ENV_VARS` |An array of values for environment variables to set in the build process, listed in the `KEY=VALUE` format. |`array` |
+|`CONTEXT` |Path to the directory within the `source` workspace to use as the context. |`string` |`.`
+|`STORAGE_DRIVER` |Set the Buildah storage driver to reflect the settings of the current cluster node settings. |`string` |`vfs`
+|`FORMAT` |The format of the container to build, either `oci` or `docker`. |`string` |`oci`
+|`BUILD_EXTRA_ARGS` |Extra parameters for the `build` command when building the image. |`string` |
+|`PUSH_EXTRA_ARGS` |Extra parameters for the `push` command when pushing the image. |`string` |
+|`SKIP_PUSH` |Skip pushing the image to the container registry. |`string` |`false`
+|`TLS_VERIFY` |The TLS verification flag, normally `true`. |`string` |`true`
+|`VERBOSE` |Turn on verbose logging; all commands executed are added to the log. |`string` |`false`
+|`VERSION` |The tag of the image stream, which corresponds to the language version. |`string` |`latest`
+|===
+
+.Supported workspaces for the `s2i-{lang}` task
+[cols="1,3",options="header"]
+|===
+| Workspace | Description
+|`source` |The application source code, which is the build context for the S2I workflow.
+|`dockerconfig` |An optional workspace for providing a `.docker/config.json` file that Buildah uses to access the container registry. Place the file at the root of the workspace with the name `config.json` or `.dockerconfigjson`.
+|===
+
+.Results that the `s2i-{lang}` task returns
+[cols="1,1,2",options="header"]
+|===
+| Result | Type | Description
+|`IMAGE_URL` |`string` |The fully qualified name of the image that was built.
+|`IMAGE_DIGEST` |`string` |Digest of the image that was built.
+|===
+
+.Changes from the `s2i-{lang}` `ClusterTask`
+
+* Several new parameters were added.
+* The `BASE_IMAGE` parameter was removed.
+* The parameter name `PATH_CONTEXT` was changed to `CONTEXT`.
+* The parameter name `TLS_VERIFY` was changed to `TLSVERIFY`.
+* The `IMAGE_URL` result was added.
+
+:lang:
```

Comment:
The syntax to undefine is:

```suggestion
:!lang:
```

But it doesn't seem to be hurting anything right now. In the interest of time, I'm going to merge this PR as is, if you can please update this in a future PR

Reply:
thanks!